__TESTS__/unit/actions/Overlay.test.ts

```diff
@@ -0,0 +1,30 @@
+import {fill} from "../../../src/actions/resize/Resize";
+import {auto, best} from "../../../src/actions/quality/Quality";
+import {imageLayer} from "../../../src/actions/layers/overlay";
+import {imageSource} from "../../../src/actions/layers/Sources";
+import TransformableImage from "../../../src/transformation/TransformableImage";
+
+describe('Tests for overlay actions', () => {
+  it('Parses an overlay with an imageSource', () => {
+    const tImage = new TransformableImage('sample');
+    tImage
+      .setConfig({
+        cloud: {
+          cloudName:'demo'
+        }
+      })
+      .resize(fill(200, 200))
+      .quality(best())
+      .overlay(
+        // imageLayerAction is what registers on the transformation(which has toString() method)
+        // imageSource is the parameter for imageLayerAction
+        imageLayer( imageSource('sample')
+          .resize(fill(50, 50))
+          .quality(auto())
+        )
+      );
+
+    expect(tImage.toURL())
+      .toBe('http://res.cloudinary.com/demo/image/upload/c_fill,w_200,h_200/q_auto:best/l_sample,c_fill,w_50,h_50,q_auto/sample');
```

Comment:
I know this isn't aligned with the spec but still i'd like to point our that eventually here you need to add fl_layer_apply at the end of the layer and move the actions to different components